### ISSUE TITLE
Online DPO: `objective/kl` and `objective/entropy` include padding tokens

### ISSUE THREAD

**Issue by mmjerge** [huggingface/trl#6803]
### Describe the bug

In Online DPO, `objective/kl`, `objective/entropy`, `objective/non_score_reward` and `objective/rlhf_reward` sum per-token logprobs over **all** completion positions, including padding after EOS.

`trl/experimental/online_dpo/online_dpo_trainer.py:1252-1266`
```python
kl = logprobs - ref_logprobs
mean_kl = kl.sum(1).mean()
...
non_score_reward = (-self.beta * kl).sum(1)
...
mean_entropy = -logprobs.sum(1).mean()
```

`logprobs` comes from `_forward`, which returns a value at every completion position. The loss path in the same function masks these correctly:

`online_dpo_trainer.py:1217-1221`
```python
padding_mask = ~completion_mask.bool()
cr_padding_mask = padding_mask[cr_indices]
cr_logprobs_sum = (cr_logprobs * ~cr_padding_mask).sum(1)
```

but `completion_mask` is never applied on the stats path. The reported values therefore scale with how much padding a batch happens to carry, making them non-comparable across batches and across `max_new_tokens` settings. Training is unaffected — this is logging only.

Note the symptom differs from the PPO case below: Online DPO has no `INVALID_LOGPROB` sentinel, so the pad positions contribute ordinary (finite, negative) log-softmax values rather than `+1.0`. Entropy does not go impossibly negative here; it is simply inflated.

### The two sibling trainers were already fixed

- PPO: fixed in #6121 (merged 2026-07-27), which multiplies by `(~padding_mask).float()` before summing.
- XPO / Nash-MD: mask upstream with `masked_fill(padding_mask, 0.0)` (`xpo_trainer.py:298-301`) so their `objective/kl` and `objective/entropy` at `xpo_trainer.py:398-407` are already clean.

Online DPO appears to be the remaining unfixed copy of this pattern.

### Suggested fix

Apply `completion_mask` before summing, e.g.

```python
kl = (logprobs - ref_logprobs) * completion_mask
mean_kl = kl.sum(1).mean()
...
mean_entropy = -(logprobs * completion_mask).sum(1).mean()
```

Happy to open a PR aligning it with the PPO/XPO/Nash-MD copies.

### System Info

`trl` @ main (f724602), `transformers` 5.9.0, `torch` 2.12.0.

**Comment by mmjerge** [comment]
Duplicate of #6809 (which also covers the PPO/XPO/GRPO metric variants) — closing to consolidate. Apologies for the noise; see my reply on #6800 for how this happened.
